From d12808dcd8e98854af1970ffc5a68527e307ff36 Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <martinez.javier@gmail.com>
Date: Wed, 5 Oct 2011 19:13:05 +0200
Subject: [PATCH 4/9] tvp5150: Add constants for PAL and NTSC video standards

Signed-off-by: Javier Martinez Canillas <martinez.javier@gmail.com>
Signed-off-by: Enrico Butera <ebutera@users.berlios.de>
---
 include/media/tvp5150.h |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/include/media/tvp5150.h b/include/media/tvp5150.h
index 72bd2a2..8991e3b 100644
--- a/include/media/tvp5150.h
+++ b/include/media/tvp5150.h
@@ -30,5 +30,11 @@
 #define TVP5150_NORMAL       0
 #define TVP5150_BLACK_SCREEN 1
 
+/* Number of pixels and number of lines per frame for different standards */
+#define NTSC_NUM_ACTIVE_PIXELS		(720)
+#define NTSC_NUM_ACTIVE_LINES		(525)
+#define PAL_NUM_ACTIVE_PIXELS		(720)
+#define PAL_NUM_ACTIVE_LINES		(625)
+
 #endif
 
-- 
1.7.5.4

