From 1615e9317858576d83810eccba49c0bc8147a460 Mon Sep 17 00:00:00 2001
From: Enrico Butera <ebutera@users.berlios.de>
Date: Sun, 9 Oct 2011 12:46:36 +0200
Subject: [PATCH 6/9] tvp5150: compile fixes and added missing entity_cleanup

Signed-off-by: Enrico Butera <ebutera@users.berlios.de>
---
 drivers/media/video/tvp5150.c |   14 +++++++++++---
 1 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/media/video/tvp5150.c b/drivers/media/video/tvp5150.c
index 1ff0eb5..e774a25 100644
--- a/drivers/media/video/tvp5150.c
+++ b/drivers/media/video/tvp5150.c
@@ -15,6 +15,7 @@
 #include <media/v4l2-chip-ident.h>
 #include <media/v4l2-ctrls.h>
 #include <media/v4l2-subdev.h>
+#include <media/v4l2-mediabus.h>
 
 #include "tvp5150_reg.h"
 
@@ -1218,6 +1219,11 @@ static int tvp5150_open(struct v4l2_subdev *subdev, struct v4l2_subdev_fh *fh)
 	return 0;
 }
 
+static int tvp5150_close(struct v4l2_subdev *subdev, struct v4l2_subdev_fh *fh)
+{
+	return 0;
+}
+
 static const struct v4l2_ctrl_ops tvp5150_ctrl_ops = {
 	.s_ctrl = tvp5150_s_ctrl,
 };
@@ -1240,8 +1246,9 @@ static const struct v4l2_subdev_core_ops tvp5150_core_ops = {
 #endif
 };
 
-static struct v4l2_subdev_file_ops tvp5150_subdev_file_ops = {
+static struct v4l2_subdev_internal_ops tvp5150_subdev_internal_ops = {
 	.open		= tvp5150_open,
+	.close		= tvp5150_close,
 };
 
 static const struct v4l2_subdev_tuner_ops tvp5150_tuner_ops = {
@@ -1311,7 +1318,6 @@ static struct v4l2_subdev_pad_ops tvp5150_pad_ops = {
 
 static const struct v4l2_subdev_ops tvp5150_ops = {
 	.core = &tvp5150_core_ops,
-	.file	= &tvp5150_subdev_file_ops,
 	.tuner = &tvp5150_tuner_ops,
 	.video = &tvp5150_video_ops,
 	.vbi = &tvp5150_vbi_ops,
@@ -1365,6 +1371,7 @@ static int tvp5150_probe(struct i2c_client *c,
 	}
 
 	core->sd.flags |= V4L2_SUBDEV_FL_HAS_DEVNODE;
+	core->sd.internal_ops = &tvp5150_subdev_internal_ops;
 	core->norm = V4L2_STD_ALL;	/* Default is autodetect */
 	core->input = TVP5150_COMPOSITE1;
 	core->enable = 1;
@@ -1391,7 +1398,7 @@ static int tvp5150_probe(struct i2c_client *c,
 	if (debug > 1)
 		tvp5150_log_status(sd);
 
-	core->pad.flags = MEDIA_PAD_FLAG_OUTPUT;
+	core->pad.flags = MEDIA_PAD_FL_SOURCE;
 	ret = media_entity_init(&core->sd.entity, 1, &core->pad, 0);
 	if (ret < 0)
 		kfree(core);
@@ -1410,6 +1417,7 @@ static int tvp5150_remove(struct i2c_client *c)
 
 	v4l2_device_unregister_subdev(sd);
 	v4l2_ctrl_handler_free(&decoder->hdl);
+	media_entity_cleanup(&sd->entity);
 	kfree(to_tvp5150(sd));
 	return 0;
 }
-- 
1.7.5.4

